presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/DomainConverter.java

```diff
@@ -49,46 +50,46 @@ private DomainConverter() {}
-                        }
+        return TupleDomain.withColumnDomains(Maps.transformValues(
+                tupleDomain.getDomains().get(),
+                DomainConverter::translateDomain));
+    }
 
-                        Marker high = range.getHigh();
-                        if (high.getValueBlock().isPresent()) {
-                            Block value = nativeValueToBlock(type, convertToMicros(type, (long) range.getHigh().getValue()));
-                            high = new Marker(range.getType(), Optional.of(value), range.getHigh().getBound());
-                        }
+    private static Domain translateDomain(Domain domain)
+    {
+        ValueSet valueSet = domain.getValues();
+        Type type = domain.getType();
+        if (type instanceof TimestampType || type instanceof TimestampWithTimeZoneType || type instanceof TimeType || type instanceof TimeWithTimeZoneType) {
+            if (valueSet instanceof EquatableValueSet) {
+                EquatableValueSet equatableValueSet = (EquatableValueSet) valueSet;
+                Set<ValueEntry> values = equatableValueSet.getEntries().stream()
+                        .map(value -> ValueEntry.create(value.getType(), convertToMicros(type, (long) value.getValue())))
+                        .collect(toImmutableSet());
+                valueSet = new EquatableValueSet(equatableValueSet.getType(), equatableValueSet.isWhiteList(), values);
+            }
+            else if (valueSet instanceof SortedRangeSet) {
+                List<Range> ranges = new ArrayList<>();
+                for (Range range : valueSet.getRanges().getOrderedRanges()) {
+                    Marker low = range.getLow();
+                    if (low.getValueBlock().isPresent()) {
+                        Block value = nativeValueToBlock(type, convertToMicros(type, (long) range.getLow().getValue()));
+                        low = new Marker(range.getType(), Optional.of(value), range.getLow().getBound());
+                    }
 
-                        ranges.add(new Range(low, high));
+                    Marker high = range.getHigh();
+                    if (high.getValueBlock().isPresent()) {
+                        Block value = nativeValueToBlock(type, convertToMicros(type, (long) range.getHigh().getValue()));
+                        high = new Marker(range.getType(), Optional.of(value), range.getHigh().getBound());
                     }
-                    transformedValueSet = SortedRangeSet.copyOf(valueSet.getType(), ranges);
+
+                    ranges.add(new Range(low, high));
                 }
-                transformedMap.put(column, Domain.create(transformedValueSet, domain.isNullAllowed()));
+                valueSet = SortedRangeSet.copyOf(valueSet.getType(), ranges);
             }
-        });
-        return TupleDomain.withColumnDomains(transformedMap);
+        }
+        return Domain.create(valueSet, domain.isNullAllowed());
```

Comment:
It is better to avoid creating a new `Domain` object and return the original `domain` here.  It will also be more clear that  `domain` is not changed for the majority of Types.